backend/de.metas.async/src/main/java/de/metas/async/processor/impl/planner/QueueProcessorPlanner.java

```diff
@@ -0,0 +1,360 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.async.processor.impl.planner;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.async.api.WorkPackageLockHelper;
+import de.metas.async.model.I_C_Queue_WorkPackage;
+import de.metas.async.processor.IQueueProcessor;
+import de.metas.async.processor.QueuePackageProcessorId;
+import de.metas.async.processor.QueueProcessorId;
+import de.metas.async.processor.impl.AbstractQueueProcessor;
+import de.metas.lock.api.ILockManager;
+import de.metas.logging.LogManager;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.service.ISysConfigBL;
+import org.compiere.model.IQuery;
+import org.compiere.util.Env;
+import org.slf4j.Logger;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+
+public abstract class QueueProcessorPlanner implements Runnable
+{
+	private static final Logger logger = LogManager.getLogger(QueueProcessorPlanner.class);
+
+	private final static String SYSCONFIG_POLLINTERVAL = "de.metas.async.PollIntervallMillis";
+
+	private final ILockManager lockManager = Services.get(ILockManager.class);
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	private final ReentrantLock mainLock = new ReentrantLock();
+	protected final ConcurrentHashMap<QueueProcessorId, IQueueProcessor> queueProcessors;
+	protected final AtomicBoolean isRunning;
+
+	public QueueProcessorPlanner()
+	{
+		this.queueProcessors = new ConcurrentHashMap<>();
+		this.isRunning = new AtomicBoolean(false);
+	}
+
+	@Override
+	public void run()
+	{
+		if (!isRunning.get())
+		{
+			throw new AdempiereException("Planner is not running! Hint: QueueProcessorPlanner should be started via start() method!");
+		}
+
+		while (isRunning.get())
+		{
+			try
+			{
+				final boolean successfulRun = run0();
+
+				if (!successfulRun && stopOnFailedRun())
+				{
+					logger.warn("*** QueueProcessorPlanner.run() -> last run was not successful & planner = " + this.getClass().getName() + " has stopOnFailedRun policy; Shutting down...");
+					shutdown();
+				}
+				else
+				{
+					// note: we always get the new service, because things might have changed since this method started
+					final int pollIntervalMs = Services.get(ISysConfigBL.class).getIntValue(SYSCONFIG_POLLINTERVAL, 1000);
```

Comment:
extract `Services.get(ISysConfigBL.class)` to a class field

extract the whole line in some `getPollIntervalInMillis` method.

Reply:
extract `1000` to DEFAULT_POLLINTERVAL_MILLIS constant